libkbfs/disk_limiter.go

```diff
@@ -48,24 +65,32 @@ type DiskLimiter interface {
 		journalStoredBytes, journalUnflushedBytes, journalFiles int64,
 		chargedTo keybase1.UserOrTeamID)
 
-	// beforeBlockPut is called before putting a block of the
-	// given byte and file count, both of which must be > 0. It
-	// may block, but must return immediately with a
-	// (possibly-wrapped) ctx.Err() if ctx is cancelled. The
-	// updated available byte and file count must be returned,
+	// reserveWithBackpressure is called before using disk storage of the given
+	// byte and file count, both of which must be > 0. It may block, but must
+	// return immediately with a (possibly-wrapped) ctx.Err() if ctx is
+	// cancelled. The updated available byte and file count must be returned,
 	// even if err is non-nil.
-	beforeBlockPut(ctx context.Context,
+	reserveWithBackpressure(ctx context.Context, typ diskLimitTrackerType,
 		blockBytes, blockFiles int64, chargedTo keybase1.UserOrTeamID) (
 		availableBytes, availableFiles int64, err error)
 
-	// afterBlockPut is called after putting a block of the given
-	// byte and file count, which must match the corresponding call to
-	// beforeBlockPut. putData reflects whether or not the data
-	// was actually put; if it's false, it's either because of an
-	// error or because the block already existed.
-	afterBlockPut(ctx context.Context,
-		blockBytes, blockFiles int64, putData bool,
-		chargedTo keybase1.UserOrTeamID)
+	// reserve is called by the disk block cache before using disk storage with
+	// the given byte count. It returns the total number of available bytes.
+	reserve(ctx context.Context, typ diskLimitTrackerType, blockBytes int64) (
+		availableBytes int64, err error)
+
+	// commitOrRollback is called after using disk storage of the given byte
+	// and file count, which must match the corresponding call to
+	// beforeBlockPut. putData reflects whether or not the data was actually
+	// put; if it's false, it's either because of an error or because the block
+	// already existed.
+	commitOrRollback(ctx context.Context, typ diskLimitTrackerType, blockBytes,
+		blockFiles int64, shouldCommit bool, chargedTo keybase1.UserOrTeamID)
+
+	// releaseAndCommit is called after deleting blocks for a given tracker of
+	// the given byte and file count, both of which must be >= 0.
+	releaseAndCommit(ctx context.Context, typ diskLimitTrackerType, blockBytes,
```

Comment:
Understood.  But to me, the name "reserve" implies that there needs to be a followup action, while "release" doesn't.  I don't feel strongly about it though.